### ISSUE TITLE
Text Rendering not happy when opacity != 1

### ISSUE THREAD

**Issue by danwalmsley** [AvaloniaUI/Avalonia#1092]
Here is simple XAML to repro the issue:
```
<Window xmlns="https://github.com/avaloniaui" MinWidth="400" MinHeight="300" MaxWidth="400" MaxHeight="300"
        xmlns:local="clr-namespace:TestTool;assembly=TestTool">
  <Grid>
    <Grid.Styles>
      <Style Selector="Grid#MainUi:disabled">
        <Setter Property="Opacity" Value="0.1" />
      </Style>
    </Grid.Styles>
    <Grid Name="MainUi" IsEnabled="{Binding UiEnabled, FallbackValue=False}">
      <DockPanel LastChildFill="True">
        <StackPanel Orientation="Horizontal" Gap="4" DockPanel.Dock="Bottom">
          <TextBlock Text="Status:" />
          <TextBlock Text="{Binding ConnectionStatusText, FallbackValue=Disconnected}" />
        </StackPanel>
        <DockPanel LastChildFill="True">
          <DockPanel LastChildFill="True" Margin="10" DockPanel.Dock="Left" >
            <TextBlock Text="Lighting" DockPanel.Dock="Top" FontSize="20" TextAlignment="Center" />
            <DropDown Items="{Binding LedBoards}" DockPanel.Dock="Top" Margin="4" />
            <local:LightingView />
          </DockPanel>
          <DockPanel LastChildFill="True" Margin="30 10 10 10" DockPanel.Dock="Left" >
            <TextBlock Text="Camera Position" DockPanel.Dock="Top" FontSize="20" TextAlignment="Center" />
            <StackPanel Orientation="Horizontal" Gap="20" HorizontalAlignment="Center" >
              <StackPanel.Styles>
                <Style Selector="TextBlock">
                  <Setter Property="TextAlignment" Value="Center" />
                </Style>
              </StackPanel.Styles>
              <StackPanel  Gap="4">
                <TextBlock Text="A" />
                <TextBox Text="{Binding CameraPositionA}" />
                <Button Content="GO" Command="{Binding MoveCameraACommand}" />
              </StackPanel>

              <StackPanel  Gap="4">
                <TextBlock Text="B" />
                <TextBox Text="{Binding CameraPositionB}" />
                <Button Content="GO" Command="{Binding MoveCameraBCommand}" />
              </StackPanel>

              <StackPanel Gap="4">
                <TextBlock Text="C" />
                <TextBox Text="{Binding CameraPositionC}" />
                <Button Content="GO" Command="{Binding MoveCameraCCommand}" />
              </StackPanel>
            </StackPanel>
          </DockPanel>
        </DockPanel>
      </DockPanel>
    </Grid>
    <Grid Background="Black" Opacity="0.1" IsVisible="{Binding !UiEnabled}" />
  </Grid>
</Window>
```

**Comment by danwalmsley** [comment]
![image](https://user-images.githubusercontent.com/4672627/29026095-ebe7e52a-7b71-11e7-814b-c77c8e07662c.png)

**Comment by danwalmsley** [comment]
Above image is on skia backend.

Notice also that the text is cut off (this also happens on Direct2d) full text is show if opacity is set to 1 here:

```
<Style Selector="Grid#MainUi:disabled">         <Setter Property="Opacity" Value="0.1" />
```

**Comment by jkoritzinsky** [comment]
Fixed by #1162
